tests/vm/test_gas.ml

```diff
@@ -0,0 +1,45 @@
+open Lambda_vm
+
+let counter =
+  [%lambda_vm.script
+    fun x ->
+      ( (fun f -> f f x) (fun f n ->
+            if n then
+              1L + f f (n - 1L)
+            else
+              0L),
+        (0L, 0L) )]
+
+let%expect_test _ =
+  let gas = Gas.measure () in
+  let _ =
+    Vm_test.compile_value_exn gas (Int64 0L) in
+  ();
+  print_int (Gas.current gas);
+  [%expect{| 100 |}]
```

Comment:
In general the way that the gas model is being tested here is not very good, `(Gas.is_empty gas)` should be true, which means what is parameterized is the initial_gas not the final one.

Reply:
Oh, but then I think that would be changed to a regular Alcotest testsuite instead of using ppx_expect.

Reply:
Changed to Alcotest